dune/grid/common/p2pcommunicator.hh

```diff
@@ -223,6 +220,32 @@ public:
      *  if receive buffers are known from previous run and have not changed
      *  communication could be faster */
     virtual void exchangeCached ( DataHandleInterface& ) const;
+
+  protected:
+    inline void computeDestinations( const linkage_t& linkage, vector_t& dest );
+
+    // return new tag number for the exchange messages
+    static int getMessageTag( const unsigned int increment )
+    {
+      static int tag = messagetag + 2 ;
+      // increase tag counter
+      const int retTag = tag;
+      tag += increment ;
+      // the MPI standard guaratees only up to 2^15-1
+      // this needs to be revised for the all-to-all communication
+      if( tag < 0 ) // >= 32767 )
```

Comment:
You can't say this.  There is nothing in the C++ standard that guarantees that signed integers wrap on overflow.  If you're really concerned about integer overflow, you need to say something along the lines of
```C++
if (tag > std::numeric_limits<int>::max() - increment) {
    // Reset to initial value
    tag = messagetag + 2;
}
else {
    tag += increment;
}
```